File: 1100/2193D/main.java
import java.util.*;
public class main 
{
	public static void main(String[] args)
	{
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
 
		while (t-- > 0)
		{
			int n = sc.nextInt();
			long[] a = new long[n];
			
			for (int i = 0; i < n; i++)
				a[i] = sc.nextLong();
		
			int[] b = new int[n];
			for (int i = 0; i < n; i++)
				b[i] = sc.nextInt();
 
			Arrays.sort(a);
			long max = 0;
			int used = 0;
 
			for (int i = 0; i < n; i++) 
			{
				used += b[i];
				
				if (used > n)
					break;
					
				long diff = a[n - used];
				long score = diff * (i + 1);
				
				if (score > max)
					max = score;
			}
			
			System.out.println(max);
		}
	}
}

File: 1100/2193D/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    while (t--) 
    {
        int n;
        cin >> n;

        vector<long long> a(n);
        for (int i = 0; i < n; i++)
            cin >> a[i];

        vector<int> b(n);
        for (int i = 0; i < n; i++)
            cin >> b[i];

        sort(a.begin(), a.end());

        long long maxScore = 0;
        int used = 0;

        for (int i = 0; i < n; i++) 
        {
            used += b[i];

            if (used > n) 
                break;

            long long diff = a[n - used];
            long long score = diff * (i + 1);

            maxScore = max(maxScore, score);
        }

        cout << maxScore << "\n";
    }

    return 0;
}
